src/overlays/actors/ovl_En_Fsn/z_en_fsn.c

```diff
@@ -21,130 +90,1557 @@ const ActorInit En_Fsn_InitVars = {
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE45C4.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4638.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4828.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4A1C.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4B20.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4B70.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4BF4.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4CD8.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE4D28.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/EnFsn_Init.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/EnFsn_Destroy.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/EnFsn_Update.asm")
-
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE502C.asm")
+ActorAnimationEntryS sAnimations[] = {
+    { &D_06012C34, 1.0f, 0, -1, 0, 0 },  { &D_060131FC, 1.0f, 0, -1, 0, 0 }, { &D_0600C58C, 1.0f, 0, -1, 2, 0 },
+    { &D_0600C58C, -1.0f, 0, -1, 2, 0 }, { &D_0600E3EC, 1.0f, 0, -1, 2, 0 }, { &D_0600F00C, 1.0f, 0, -1, 0, 0 },
+    { &D_0600CB3C, 1.0f, 0, -1, 2, 0 },  { &D_0600D354, 1.0f, 0, -1, 0, 0 }, { &D_060138B0, 1.0f, 0, -1, 2, 0 },
+    { &D_0601430C, 1.0f, 0, -1, 0, 0 },  { &D_0600B9D8, 1.0f, 0, -1, 2, 0 }, { &D_0600C26C, 1.0f, 0, -1, 0, 0 },
+    { &D_0600DE34, 1.0f, 0, -1, 2, 0 },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE52A4.asm")
+static ColliderCylinderInit sCylinderInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        OC2_TYPE_1,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK1,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
+    { 18, 64, 0, { 0, 0, 0 } },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE5440.asm")
+static Vec3f sShopItemPositions[3] = {
+    { -5.0f, 35.0f, -95.0f },
+    { 13.0f, 35.0f, -95.0f },
+    { 31.0f, 35.0f, -95.0f },
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fsn_0x80AE1B70/func_80AE57E8.asm")
+s32 EnFsn_TestItemSelected(GlobalContext* globalCtx) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+
+    if (msgCtx->unk12020 == 0x10 || msgCtx->unk12020 == 0x11) {
+        return CHECK_BTN_ALL(globalCtx->state.input[0].press.button, BTN_A);
+    }
+    return CHECK_BTN_ALL(globalCtx->state.input[0].press.button, BTN_A) ||
```

Comment:
Do we want to make a Macro for each controller? For example `CONTROLLER1 = &globalCtx->state.input[0]`.

Reply:
My vote is yes.

Reply:
I'm also good with this Macro. Related to this, is CONTROLLER3 ever used in MM? I was recently working on db_camera from OoT which did use `&globalCtx->state.input[2]`.